### ISSUE TITLE
machinectl start fails with Ubuntu Wily disk image - machine id fails to parse

### ISSUE THREAD

**Issue by NeilW** [systemd/systemd#2091]
I'm running an official Fedora 23 cloud image (from https://download.fedoraproject.org/pub/fedora/linux/releases/23/Cloud/x86_64/Images/Fedora-Cloud-Base-23-20151030.x86_64.raw.xz) on KVM with the systemd-nspawn@.service file suitably patched so that 'machinectl start' won't fail with /dev/loop-control permission issues.

If I run an official Ubuntu disk image for the latest release. 

```
$ sudo machinectl pull-raw https://cloud-images.ubuntu.com/releases/15.10/release/ubuntu-15.10-server-cloudimg-amd64-disk1.img wily
$ sudo machinectl start wily
```

```
systemctl status systemd-nspawn@wily.service -l
● systemd-nspawn@wily.service - Container wily
   Loaded: loaded (/usr/lib/systemd/system/systemd-nspawn@.service; disabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Thu 2015-12-03 17:31:03 UTC; 47s ago
     Docs: man:systemd-nspawn(1)
  Process: 1533 ExecStart=/usr/bin/systemd-nspawn --quiet --keep-unit --boot --link-journal=try-guest --network-veth --machine=%I (code=exited, status=1/FAILURE)
 Main PID: 1533 (code=exited, status=1/FAILURE)
   Status: "Terminating..."

Dec 03 17:31:03 srv-x84n8.gb1s.brightbox.com systemd[1]: Starting Container wily...
Dec 03 17:31:03 srv-x84n8.gb1s.brightbox.com systemd-nspawn[1533]: Failed to parse machine ID from /tmp/nspawn-root-DCpwgk/etc/machine-id: Invalid argument
Dec 03 17:31:03 srv-x84n8.gb1s.brightbox.com systemd[1]: systemd-nspawn@wily.service: Main process exited, code=exited, status=1/FAILURE
Dec 03 17:31:03 srv-x84n8.gb1s.brightbox.com systemd[1]: Failed to start Container wily.
Dec 03 17:31:03 srv-x84n8.gb1s.brightbox.com systemd[1]: systemd-nspawn@wily.service: Unit entered failed state.
Dec 03 17:31:03 srv-x84n8.gb1s.brightbox.com systemd[1]: systemd-nspawn@wily.service: Failed with result 'exit-code'.
```

**Comment by poettering** [comment]
Any idea how precisely the file /etc/machine-id looks like on that image when this fails?

**Comment by NeilW** [comment]
Ubuntu Wily 15.10 contains an empty file but exists.

```
# chroot /mnt ls -la /etc/machine-id
-r--r--r--. 1 root root 0 Jan 14 19:22 /etc/machine-id
```

**Comment by NeilW** [comment]
If you start the image under KVM instead of a container, then the machine-id is populated by systemd on boot

```
Jan 27 16:30:58 ubuntu systemd[1]: Set hostname to <ubuntu>.
Jan 27 16:30:58 ubuntu systemd[1]: Initializing machine ID from KVM UUID.
Jan 27 16:30:58 ubuntu systemd[1]: Installed transient /etc/machine-id file.
```

**Comment by poettering** [comment]
Ah, right, we should really handle that more nicely I guess, and initialize the machine id.

**Comment by poettering** [comment]
Fix waiting in #2466.
